recovery_backend/backend/configmap.go

```diff
@@ -23,21 +23,35 @@ type RecoveryBackend struct {
 	Datastore *datastore.DataStore
 }
 
-func newConfigMap(namespace, name, version string) *v1.ConfigMap {
+func (rb *RecoveryBackend) newConfigMap(namespace, name, version, smPodName string) (*v1.ConfigMap, error) {
+	smPod, err := rb.Datastore.GetPod(smPodName)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get share manager pod %v", name)
+	}
+
+	smName, ok := smPod.Labels[types.GetLonghornLabelKey(types.LonghornLabelShareManager)]
+	if !ok {
+		return nil, errors.Wrapf(err, "failed to get label %v from share manager pod %v",
```

Comment:
NIT: `smPod.Name` rather than `smName`. `smName` is empty here.

BTW, is the smPod name always the same as ShareManager CR name? If Yes, I think we can directly use the pod name and verify the ShareManager CR existence.